browser/src/control/Control.JSDialog.js

```diff
@@ -525,12 +532,36 @@ L.Control.JSDialog = L.Control.extend({
 		instance.posy = top + offsetY;
 
 		var width = instance.form.getBoundingClientRect().width;
-		var height = instance.form.getBoundingClientRect().height;
+		var autoFilterBottom = instance.posy + document.querySelector('.main-nav').clientHeight + document.getElementById('toolbar-wrapper').clientHeight + instance.form.getBoundingClientRect().height;
```

Comment:
Can't we just use `instance.form.getBoundingClientRect().bottom`  ? better to let browser calculate that, than do it on your own. Especially that now it introduces dependency on "main-nav" and "toolbar-wrapper". We will want to change classic mode in the future - it will cause problems.

Reply:
100 % agreed. I have added too much complicity. Surly update this thanks :)

Reply:
things like: `document.querySelector('sss')` can fail when we remove some element in the future. It will cause error and half-baked dialog/popup. Please first get query result into variable, check if exists and then use. If not exist - handle somehow that case.

Reply:
But can we avoid using `document.querySelector` at all?
You are interested in map's top edge. Can't you access it using some variables and it's members? it's more safe to use and reliable (from: `this.map.xxxx`)